application/src/main/java/uk/gov/hmcts/dm/repository/RepositoryFinder.java

```diff
@@ -31,9 +32,9 @@
 
         Iterator<Class<?>> it = repositories.iterator();
         while (it.hasNext()) {
-            Object repository = repositories.getRepositoryFor(domainClass);
-            if (repository != null && repository instanceof CrudRepository) {
-                return (CrudRepository<Object, Serializable>)repository;
+            Optional<Object> repository = repositories.getRepositoryFor(domainClass);
```

Comment:
I get a compilation error on this line, because getRepositoryFor does not return optional